testsuite/classlibrary/TestNodeProxy.sc

```diff
@@ -40,4 +40,137 @@ TestNodeProxy : UnitTest {
 	}
 
+	test_asCode_basic {
+		var codeString = "a = NodeProxy.new(Server.fromName( 'TestNodeProxy' ));\n";
+
+		this.assertEquals(proxy.asCode, codeString,
+			"asCode-posting basic nodeproxy should post valid source code.");
+	}
+
+	test_asCode_single {
+		var codeString = "a = NodeProxy.new(Server.fromName( 'TestNodeProxy' )).source_({ DC.ar });\n";
+		proxy.source = { DC.ar };
+
+		this.assertEquals(proxy.asCode, codeString,
+			"asCode-posting single-source nodeproxy should post simple source form.");
+	}
+
+	test_asCode_multi {
+		var asCodeString =
+		"(\n"
+		"a = NodeProxy.new;\n"
+		"a[5] = { DC.ar };\n"
+		"a[10] = { DC.ar(0.01) };\n"
+		"\n"
+		");\n";
+		proxy[5] = { DC.ar };
+		proxy[10] = { DC.ar(0.01) };
+		this.assertEquals(proxy.asCode.postcs, asCodeString,
+			"asCode-posting multi-source nodeproxy asCode should post all its sources.");
+	}
+
+	test_asCode_settings {
+		var codeString =
+		"(\n"
+		"a = NodeProxy.new(Server.fromName( 'TestNodeProxy' ));\n"
+		"a.set('freq', 440);\n"
+		");\n";
+		proxy.set('freq', 440);
+
+		this.assertEquals(proxy.asCode, codeString,
+			"asCode-posting nodeproxy with settings should post these correctly."
+		);
+	}
+
+	// this one needs the server booted ...
+	test_asCode_playState {
+		var codeString =
+		"(\n"
+		"a = NodeProxy.new(Server.fromName( 'TestNodeProxy' ));\n"
+		"a.play(\n"
+		"	out: 8, \n"
+		"	vol: 0.25\n"
+		"\n"
+		");\n"
+		");\n";
+
+		this.bootServer(server);
+
+		proxy.play(8, 2, vol: 0.25);
+
+		server.sync; // avoids "FAILURE IN SERVER /g_new Group 1 not found"
+
+		this.assertEquals(proxy.asCode, codeString,
+			"asCode-posting nodeproxy with settings should post these correctly.",
+			onFailure: { server.quit }
+		);
+
+		server.quit;
+	}
+
+	test_asCode_single_ndef {
+		var codeString = "Ndef('x', { DC.ar });\n";
+		proxy = Ndef(\x, { DC.ar });
+
+		this.assertEquals(proxy.asCode, codeString,
+			"asCode-posting single-source Ndef should post correctly."
+		);
+
+		Ndef(\x).free.clear;
+	}
+
+	test_asCode_multi_ndef {
+		var codeString =
+		"(\n"
+		"Ndef('x')[5] = { DC.ar };\n"
+		"Ndef('x')[10] = { DC.ar(0.01) };\n"
+		"\n"
+		");\n";
+
+		proxy = Ndef(\x);
+		Ndef(\x)[5] = { DC.ar };
+		Ndef(\x)[10] = { DC.ar(0.01) };
+
+		this.assertEquals(proxy.asCode, codeString,
+			"asCode-posting single-source Ndef should post correctly."
+		);
+
+		Ndef(\x).free.clear;
+	}
+
+	test_asCode_single_inProxySpace {
+		var codeString = "~a = { DC.ar };\n";
+		var p = ProxySpace.push;
+		p[\a] = { DC.ar };
+		p[\a].asCode.cs;
+
+		this.assertEquals(p[\a].asCode(envir: p), codeString,
+			"asCode-posting single-source proxy should post by key in pushed proxyspace."
+		);
+
+		p[\a].free;
+		p.pop;
+	}
+
+	test_asCode_multi_inProxySpace {
+		var codeString =
+		"(\n"
+		"~a[5] = { DC.ar };\n"
+		"~a[10] = { DC.ar(0.01) };\n"
+		"\n"
+		");\n";
+
+		var p = ProxySpace.push;
+		p[\a][5] = { DC.ar };
+		p[\a][10] = { DC.ar(0.01) };
+		p[\a].asCode.postcs;
```

Comment:
remove postcs

Reply:
ah thanks, removed - also the postcs in line 68.